### ISSUE TITLE
IntegrityError when submitting a post with same title+language combo

### ISSUE THREAD

**Issue by czpython** [aldryn/aldryn-newsblog#221]
```
Environment:


Request Method: POST
Request URL: https://paulosproject.ch/en/admin/aldryn_newsblog/article/add/?_changelist_filters=app_config__id__exact%3D2&language=de

Django Version: 1.6.5
Python Version: 2.7.3
Installed Applications:
['django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.sites',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.sitemaps',
 'django.contrib.gis',
 'djangocms_admin_style',
 'admin_shortcuts',
 'googleappsauth',
 'django.contrib.admin',
 'localflavor',
 'cmsplugin_filer_file',
 'cmsplugin_filer_image',
 'cmsplugin_filer_folder',
 'djangocms_text_ckeditor',
 'djangocms_link',
 'djangocms_flash',
 'djangocms_snippet',
 'aldryn_boilerplates',
 'aldryn_snake',
 'aldryn_style',
 'aldryn_grid_bootstrap',
 'aldryn_accordion',
 'aldryn_button',
 'aldryn_common',
 'aldryn_faq',
 'aldryn_forms',
 'aldryn_gallery',
 'aldryn_gallery_timed',
 'aldryn_inpage_navigation',
 'aldryn_search',
 'aldryn_segmentation',
 'aldryn_timed',
 'aldryn_video',
 'commonsearch',
 'aldryn_categories',
 'aldryn_people',
 'aldryn_newsblog',
 'parler',
 'gunicorn',
 'compressor',
 'south',
 'easy_thumbnails',
 'login_as',
 'standard_form',
 'divio_styleguide',
 'django_commontranslations',
 'captcha',
 'crispy_forms',
 'django_select2',
 'django_tablib',
 'haystack',
 'hvad',
 'location_field',
 'absolute',
 'linguo',
 'adminsortable',
 'sortedm2m',
 'taggit',
 'url_tools',
 'djcelery',
 'redisboard',
 'health_check',
 'health_check_celery3',
 'health_check_db',
 'health_check_cache',
 'health_check_storage',
 'dealer',
 'cms',
 'mptt',
 'menus',
 'sekizai',
 'filer',
 'reversion',
 'core',
 'activity_tracker',
 'channels',
 'cms_extensions',
 'content_plugins',
 'content_replacers',
 'coverage',
 'dynforms',
 'featuredsearch',
 'feedback',
 'generic_object',
 'linksearch',
 'homepage',
 'migration_scripts',
 'multitooltips',
 'navigation_menus',
 'replacer',
 'search',
 'products',
 'polls',
 'site_sections',
 'shortnumbers',
 'tabs',
 'maintenance',
 'user_alerts',
 'user_widgets',
 'raven.contrib.django.raven_compat',
 'raven.contrib.django.raven_compat']
Installed Middleware:
['core.middleware.BasicAuthMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.transaction.TransactionMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.middleware.locale.LocaleMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'site_sections.middleware.SiteSectionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.middleware.http.ConditionalGetMiddleware',
 'cms.middleware.page.CurrentPageMiddleware',
 'cms.middleware.toolbar.ToolbarMiddleware',
 'cms.middleware.user.CurrentUserMiddleware',
 'aliases.middleware.AliasRedirectionMiddleware',
 'maintenance.middleware.MaintenanceMiddleware']


Traceback:
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/django/core/handlers/base.py" in get_response
  112.                     response = wrapped_callback(request, *callback_args, **callback_kwargs)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/newrelic-2.34.0.29/newrelic/hooks/framework_django.py" in wrapper
  497.                 return wrapped(*args, **kwargs)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/django/contrib/admin/options.py" in wrapper
  432.                 return self.admin_site.admin_view(view)(*args, **kwargs)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/django/utils/decorators.py" in _wrapped_view
  99.                     response = view_func(request, *args, **kwargs)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/django/views/decorators/cache.py" in _wrapped_view_func
  52.         response = view_func(request, *args, **kwargs)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/django/contrib/admin/sites.py" in inner
  198.             return view(request, *args, **kwargs)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/reversion/revisions.py" in do_revision_context
  297.                 return func(*args, **kwargs)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/aldryn_newsblog/admin.py" in add_view
  129.         return super(ArticleAdmin, self).add_view(request, *args, **kwargs)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/django/utils/decorators.py" in _wrapper
  29.             return bound_func(*args, **kwargs)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/django/utils/decorators.py" in _wrapped_view
  99.                     response = view_func(request, *args, **kwargs)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/django/utils/decorators.py" in bound_func
  25.                 return func(self, *args2, **kwargs2)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/django/db/transaction.py" in inner
  371.                 return func(*args, **kwargs)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/django/contrib/admin/options.py" in add_view
  1131.                 self.save_model(request, new_object, form, False)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/django/contrib/admin/options.py" in save_model
  860.         obj.save()
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/aldryn_newsblog/models.py" in save
  239.             return super(Article, self).save(*args, **kwargs)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/parler/models.py" in save
  560.         self.save_translations(*args, **kwargs)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/parler/models.py" in save_translations
  615.                 self.save_translation(translation, *args, **kwargs)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/parler/models.py" in save_translation
  643.             translation.save(*args, **kwargs)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/django/db/models/base.py" in save
  545.                        force_update=force_update, update_fields=update_fields)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/parler/models.py" in save_base
  807.         super(TranslatedFieldsModel, self).save_base(raw=raw, using=using, **kwargs)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/django/db/models/base.py" in save_base
  573.             updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/django/db/models/base.py" in _save_table
  654.             result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/django/db/models/base.py" in _do_insert
  687.                                using=using, raw=raw)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/django/db/models/manager.py" in _insert
  232.         return insert_query(self.model, objs, fields, **kwargs)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/django/db/models/query.py" in insert_query
  1514.     return query.get_compiler(using=using).execute_sql(return_id)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/django/db/models/sql/compiler.py" in execute_sql
  903.             cursor.execute(sql, params)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/django/db/backends/util.py" in execute
  69.             return super(CursorDebugWrapper, self).execute(sql, params)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/django/db/backends/util.py" in execute
  53.                 return self.cursor.execute(sql, params)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/django/db/utils.py" in __exit__
  99.                 six.reraise(dj_exc_type, dj_exc_value, traceback)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/django/db/backends/util.py" in execute
  53.                 return self.cursor.execute(sql, params)
File "/home/paulosproject-dev/app/local/lib/python2.7/site-packages/newrelic-2.34.0.29/newrelic/hooks/database_dbapi2.py" in execute
  22.                         *args, **kwargs)

Exception Type: IntegrityError at /en/admin/aldryn_newsblog/article/add/
Exception Value: duplicate key value violates unique constraint "aldryn_newsblog_article_tra_language_code_5cac8eb2196f7e76_uniq"
DETAIL:  Key (language_code, slug)=(de, testtest) already exists.
```

**Comment by czpython** [comment]
So the issue happens because the code that checks if there's an existing article with given slug, uses the current language instead of the language the article is being created on.
This means that if I create an article in german with the same title as another article in german **while being on the english version of admin**, [this](https://github.com/aldryn/aldryn-newsblog/blob/develop/aldryn_newsblog/models.py#L237) check will fail to detect duplicate because it uses the current thread language (english) instead of the language I'm creating the article on.

So once this check fails then we get the IntegrityError.

**Comment by mkoistinen** [comment]
Awesome work, thanks @czpython.

**Comment by czpython** [comment]
Fixed by d14f8e277ebc7a0e8955df8f7b8a0ac19c8da4f0
